Tasks/Common/azure-arm-rest/azure-app-service.ts

```diff
@@ -0,0 +1,185 @@
+import msRestAzure = require('./azure-arm-common');
+import webClient = require('./webClient');
+import azureServiceClient = require('./AzureServiceClient');
+import tl = require('vsts-task-lib/task');
+import Q = require('q');
+import * as querystring from "querystring";
+var parseString = require('xml2js').parseString;
+
+export class AzureAppService {
+    private appServiceName: string;
+    private resourceGroupName: string;
+    private slotName: string;
+    private endpoint: any;
+    private azureServiceClient: azureServiceClient.ServiceClient; 
+    private credentials: msRestAzure.ApplicationTokenCredentials;
+    private appDetails: Promise<any>;
+
+    constructor(endpoint: any, name: string, resourceGroup: string, slot: string) {
+
+        this.appServiceName = name;
+        this.resourceGroupName = resourceGroup;
+        this.credentials = new msRestAzure.ApplicationTokenCredentials(endpoint["servicePrincipalClientID"],
+            endpoint["tenantID"], endpoint["servicePrincipalKey"], endpoint["url"], 
+            endpoint["envAuthUrl"], endpoint["activeDirectoryResourceId"], endpoint["environment"].toLowerCase() == 'azurestack'
+        );
+        this.azureServiceClient = new azureServiceClient.ServiceClient(this.credentials, endpoint["subscriptionId"]);
+        this.slotName = (slot && slot.toLowerCase() == 'production') ? null : slot;
+    }
+
+    public async getResourceGroupName() {
+        var dataDeferred = Q.defer<string>();
+        if(this.resourceGroupName) {
+            console.log("getResourceGroupName in cache");
```

Comment:
its better to check if(!!this.resourceGroupName).

Reply:
sure